File: app/src/main/java/com/jjf/template/ui/MainFragment.java
package com.jjf.template.ui;


import android.view.KeyEvent;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.jjf.template.R;
import com.jjf.template.base.BaseFragment;
import com.jjf.template.di.Injectable;
import com.jjf.template.util.SoftInputUtil;
import com.jjf.template.util.lifecycle.ViewModelFactory;

import javax.inject.Inject;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;
import butterknife.BindView;
import butterknife.OnClick;

/**
 * A simple {@link Fragment} subclass.
 * create an instance of this fragment.
 */
public class MainFragment extends BaseFragment implements Injectable {
    @Inject
    ViewModelFactory mViewModelFactory;
    @BindView(R.id.input)
    EditText mInput;
    @BindView(R.id.tv_name)
    TextView mTvName;
    @BindView(R.id.iv_avatar)
    ImageView mIvAvatar;
    @BindView(R.id.tv_company)
    TextView mTvCompany;
    @BindView(R.id.tv_repos_url)
    TextView mTvReposUrl;
    private UserViewModel mUserViewModel;

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_main;
    }

    @Override
    protected void initView(View view) {
        mUserViewModel = ViewModelProviders.of(this, mViewModelFactory).get(UserViewModel.class);

        observeResource(mUserViewModel.user, user -> {
            if (user != null) {
                Glide.with(this).load(user.getAvatarUrl()).into(mIvAvatar);
                mTvName.setText(user.getName());
                mTvCompany.setText(user.getCompany());
                mTvReposUrl.setText(user.getReposUrl());
            }
        });

        mInput.setOnEditorActionListener(new TextView.OnEditorActionListener() {
            @Override
            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
                if (actionId == EditorInfo.IME_ACTION_SEARCH) {
                    doSearch(v.getText().toString());
                    return true;
                } else {
                    return false;
                }
            }
        });
    }


    private void doSearch(String login) {
        if (getActivity() != null) {
            SoftInputUtil.INSTANCE.hideSoftInput(getActivity(), mInput);
        }
        mUserViewModel.setLogin(login);
    }

    @Override
    protected void onLazyLoad() {

    }

    @OnClick(R.id.iv_avatar)
    public void onViewClicked(View view) {
        navController().navigate(MainFragmentDirections.showResp(mInput.getText().toString()));
    }
}

File: app/src/main/java/com/jjf/template/ui/MainFragment.kt
package com.jjf.template.ui


import android.os.Bundle
import android.view.inputmethod.EditorInfo
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import com.bumptech.glide.Glide
import com.jjf.template.BaseFragment
import com.jjf.template.R
import com.jjf.template.di.Injectable
import com.jjf.template.util.SoftInputUtil
import com.jjf.template.util.findNavController
import com.jjf.template.util.lifecycle.ViewModelFactory
import com.jjf.template.util.viewModelProvider
import kotlinx.android.synthetic.main.fragment_main.*
import javax.inject.Inject

/**
 * A simple [Fragment] subclass.
 * create an instance of this fragment.
 */
 class MainFragment : BaseFragment(), Injectable {

    override fun getLayoutId(): Int = R.layout.fragment_main

    @Inject
    lateinit var mViewModelFactory: ViewModelFactory

    lateinit var mUserViewModel: UserViewModel

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        mUserViewModel = viewModelProvider(mViewModelFactory)
        mUserViewModel.user.observe(this, Observer { userResource->
            val user = userResource?.data
            Glide.with(this).load(user?.avatarUrl).into(iv_avatar)
            tv_name.text = user?.name
            tv_company.text = user?.company
            tv_repos_url.text = user?.reposUrl
        })

        input.setOnEditorActionListener { v, actionId, event ->
            if (actionId == EditorInfo.IME_ACTION_SEARCH) {
                doSearch(v.text.toString())
                true
            } else {
                false
            }
        }
        iv_avatar.setOnClickListener {
            findNavController().navigate(MainFragmentDirections.showResp(input.text.toString()))
        }
    }


    private fun doSearch(login: String) {
        if (activity != null) {
            SoftInputUtil.hideSoftInput(activity!!, input)
        }
        mUserViewModel.setLogin(login)
    }

}
